metron-analytics/metron-profiler-common/src/main/java/org/apache/metron/profiler/ProfilePeriod.java

```diff
@@ -51,7 +61,7 @@ public ProfilePeriod(long epochMillis, long duration, TimeUnit units) {
       throw new IllegalArgumentException(format(
               "period duration must be greater than 0; got '%d %s'", duration, units));
     }
-
+    this.timestamp = epochMillis;
```

Comment:
I can't see where you are using this value, but maybe I am missing it.

I'd be cautious of using this value because it could be a timestamp anywhere throughout the period.  If you really need a timestamp for the period, then it might be better to use `getStartTimeMillis()`. That will give you the same timestamp for the period; no matter what initial timestamp was used to find that period.